### ISSUE TITLE
DownstreamWebApiOptions.Clone does not include CustomizeHttpRequestMessage

### ISSUE THREAD

**Issue by rvplauborg** [AzureAD/microsoft-identity-web#1970]
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

1.25.5

### Web app

Sign-in users and call web APIs

### Web API

Protected web APIs call downstream web APIs

### Token cache serialization

In-memory caches

### Description

`DownstreamWebApi.CallWebApiForUserAsync` implementation currently merges options from initial configuration and `calledApiOptionsOverride` by cloning the initial options and then invoking the override action:
```
            DownstreamWebApiOptions clonedOptions = options.Clone();
            calledApiOptionsOverride?.Invoke(clonedOptions);
```
The problem is that Clone does not include the `CustomizeHttpRequestMessage` property on `DownstreamWebApiOptions`, which means that it is just null and you thereby lose the original configuration for that property. 
If users do not want to use the strongly typed extensions because they need more control, they then cannot use the `CustomizeHttpRequestMessage` setting at all unfortunately.

### Reproduction steps

1. Initial DI configuration:
```
builder.AddDownstreamWebApi(serviceName, options =>
            {
               options.CustomizeHttpRequestMessage =
                    message =>
                    {
                        message.Headers.Add("key", "value");
                    };
            });
```
2. Inject downstream web api, and use it:
```
await _downstreamWebApi.CallWebApiForUserAsync(ServiceName(), null, options =>
            {
                // Here the options.CustomizeHttpRequestMessage  is null unfortunately
                options.RelativePath = "some-url";
                options.HttpMethod = HttpMethod.Get;
            });
```

### Relevant code snippets

```csharp
Included in repro steps
```

### Expected behavior

Expected the `CustomizeHttpRequestMessage` setting to behave as all other settings and not be cleared.

**Comment by jmprieur** [comment]
Thanks @vplauborg

**Comment by rvplauborg** [comment]
Thanks for merging my fix so quickly. When do you reckon the change will be released just out of curiosity?

**Comment by jmprieur** [comment]
We'll try to release it tomorrow, @rvplauborg

**Comment by rvplauborg** [comment]
@jmprieur amazing. Thanks!

**Comment by jennyf19** [comment]
Included in [1.25.8](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.25.8) thanks for the contribution @rvplauborg 
We will also include this in 2.x-preview in the next release, either today or tomorrow.
